htdocs/core/lib/tax.lib.php

```diff
@@ -599,16 +599,14 @@ function tax_by_rate($type, $db, $y, $q, $date_start, $date_end, $modetax, $dire
 		$sql .= " f.".$invoicefieldref." as facnum, f.type, f.total_ttc as ftotal_ttc, f.datef, s.nom as company_name, s.rowid as company_id,";
 		$sql .= " p.rowid as pid, p.ref as pref, p.fk_product_type as ptype,";
 		$sql .= " 0 as payment_id, 0 as payment_amount";
-		$sql .= " FROM ".MAIN_DB_PREFIX.$invoicetable." as f,";
-		$sql .= " ".MAIN_DB_PREFIX."societe as s,";
-		$sql .= " ".MAIN_DB_PREFIX.$invoicedettable." as d";
+		$sql .= " FROM ".MAIN_DB_PREFIX.$invoicetable." as f";
+		$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."societe as s ON s.rowid = f.fk_soc";
```

Comment:
Why using a left join instead of default join ?

Reply:
A left join is normally faster than a default join. This is optimization. Do you want us to leave the default join?

Reply:
No, this is the opposite. A LEFT JOIN is nearly often slower that an INNER JOIN. On a left join, you say to the database, I want the record on second table even if it does not exists. It result in the execution plan of the database (you can get it by doing "EXPLAIN PLAN sqlrequest"), that the database will not try to use matching on index on both side, but will often make a full scan (on the right or on the left side, this is very very slow).
When you are sure there is a child (due to context rules), you must always use a strict join. Left join must be used only when a child may not exists and we want, despite of this, the parent (this is a different need and result, and it is slower).

Reply:
@eldy Done. Thanks for the explanation.